File: sdk/src/main/java/com/meniga/sdk/webservices/requests/CreateChallenge.java
package com.meniga.sdk.webservices.requests;

import org.joda.time.DateTime;

/**
 * Copyright 2017 Meniga Iceland Inc.
 */
public class CreateChallenge extends QueryRequestObject {

	public String title;
	public String description;
	public DateTime startDate;
	public DateTime endDate;
	public Long iconId;
	public CreateChallengeTypeData typeData;

	@Override
	public long getValueHash() {
		int result = title != null ? title.hashCode() : 0;
		result = 31 * result + (description != null ? description.hashCode() : 0);
		result = 31 * result + (startDate != null ? startDate.hashCode() : 0);
		result = 31 * result + (endDate != null ? endDate.hashCode() : 0);
		result = 31 * result + (iconId != null ? iconId.hashCode() : 0);
		result = 31 * result + (typeData != null ? typeData.hashCode() : 0);
		return result;
	}

}

File: sdk/src/main/java/com/meniga/sdk/webservices/challenge/CreateChallenge.kt
/*
 * Copyright 2017-2018 Meniga Iceland Inc.
 */
package com.meniga.sdk.webservices.challenge

import com.meniga.sdk.helpers.MenigaDecimal
import com.meniga.sdk.models.categories.enums.CategoryType
import com.meniga.sdk.models.challenges.enums.ChallengeInterval
import com.meniga.sdk.webservices.requests.QueryRequestObject

import org.joda.time.DateTime

internal data class CreateChallenge(
        @JvmField var title: String,
        @JvmField var description: String,
        @JvmField var startDate: DateTime,
        @JvmField var endDate: DateTime? = null,
        @JvmField var iconUrl: String? = null,
        @JvmField var typeData: CreateChallengeTypeData
) : QueryRequestObject() {

    override fun getValueHash(): Long = hashCode().toLong()

    internal data class CreateChallengeTypeData(
            var categoryIds: List<Long>? = null,
            var categoryType: CategoryType? = null,
            var targetAmount: MenigaDecimal? = null,
            var metaData: String? = null,
            var recurringInterval: ChallengeInterval? = null
    )
}
